src/webots/nodes/WbMotor.cpp

```diff
@@ -313,8 +313,10 @@ void WbMotor::updateSound() {
     const QString extension = completeUrl.mid(completeUrl.lastIndexOf('.') + 1).toLower();
     if (WbUrl::isWeb(completeUrl))
       mSoundClip = WbSoundEngine::sound(WbNetwork::instance()->get(completeUrl), extension);
-    else
+    else if (!(completeUrl.isEmpty() || completeUrl == WbUrl::missingTexture()))
```

Comment:
Why does the motor need to check for missing texture?

Reply:
This is in the case the user inputs an invalid .png or .jpg file in the `sound` field. This would put the `missing_texture.png` URL in `completeUrl`. If you don't add this condition here, warnings will be triggered but the `mSoundClip` won't be updated to NULL, which would cause the sound to still work even if an invalid file was given in the field.

Reply:
Maybe we should add a comment in the code explaining this, as it is not obvious.